test/jdk/java/util/spi/ToolProviderDescriptionTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8287121
+ * @summary test ToolProvider's description API
+ * @run main/othervm ToolProviderDescriptionTest
+ */
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.spi.ToolProvider;
+import java.util.stream.StreamSupport;
+
+public class ToolProviderDescriptionTest {
+    public static void main(String... args) throws Exception {
+        List<String> descriptions = listToolDescriptions();
+        if (descriptions.isEmpty()) {
```

Comment:
Hello @sormuras,
Won't this condition always be "false", because from what I see in this test code, the `descriptions` list will never be empty since in the `describeTool` method of this test we always return a `String` instance to be added to the `List`, even if there's no `description` for the `ToolProvider` instance.

Reply:
The list will be empty, if no `ToolProvider` service is observable at all. For example, when running `java` with `--limit-module java.base` - which doesn't provide an implemention of `ToolProvider` as of today.

This test checks that all observable tools don't throw on calling their `name()` and `description()` accessors.

This test does not care for proper results being returned.